src/components/AddNewBook.js

```diff
@@ -1,18 +1,46 @@
 import React from 'react';
+import { useDispatch } from 'react-redux/es/hooks/useDispatch';
+import { v4 } from 'uuid';
+import { addBook } from '../redux/books/books';
 
-class AddNewBook extends React.PureComponent {
-  render() {
-    return (
-      <div className="add-book-container">
-        <h2>ADD NEW BOOK</h2>
-        <form>
-          <input type="text" placeholder="Book Title" />
-          <input type="text" placeholder="Author" />
-          <button type="submit">ADD BOOK</button>
-        </form>
-      </div>
-    );
-  }
-}
+const AddNewBook = () => {
+  const dispatch = useDispatch();
+
+  const handleAddSubmit = (e) => {
+    e.preventDefault();
+
+    const alertMessage = document.getElementById('alert-message');
+
+    if (e.target.title.value && e.target.author.value) {
+      const newBook = {
+        title: e.target.title.value,
+        author: e.target.author.value,
+        id: v4(),
+      };
+
+      dispatch(addBook(newBook));
+
+      e.target.title.value = '';
+      e.target.author.value = '';
+    } else {
+      alertMessage.textContent = 'Please, enter a book with their title and author';
+      setTimeout(() => {
+        alertMessage.textContent = '';
+      }, 4500);
+    }
+  };
+
+  return (
+    <div className="add-book-container">
+      <h2>ADD NEW BOOK</h2>
+      <form id="main-form" onSubmit={handleAddSubmit}>
```

Comment:
- [ ] onSubmit you are running the entire component again you could just run a function instead.